SimPPS/RPDigiProducer/plugins/RPVFATSimulator.cc

```diff
@@ -0,0 +1,72 @@
+#include "SimPPS/RPDigiProducer/interface/RPVFATSimulator.h"
+#include "Geometry/VeryForwardRPTopology/interface/RPTopology.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <vector>
+#include "TRandom.h"
+#include <iostream>
+
+
+RPVFATSimulator::RPVFATSimulator(const edm::ParameterSet &params, RPDetId det_id)
+ : params_(params), det_id_(det_id)
+{
+  threshold_ = params.getParameter<double>("RPVFATThreshold");
+  dead_strip_probability_ = params.getParameter<double>("RPDeadStripProbability");
+  dead_strips_simulation_on_ = params.getParameter<bool>("RPDeadStripSimulationOn");
+  strips_no_ = RPTopology().DetStripNo();
+  verbosity_ = params.getParameter<int>("RPVerbosity");
+  links_persistence_ = params.getParameter<bool>("RPDigiSimHitRelationsPresistence");
+  
+  if(dead_strips_simulation_on_)
+    SetDeadStrips();
+}
+
+void RPVFATSimulator::ConvertChargeToHits(const simRP::strip_charge_map &signals, 
+    simRP::strip_charge_map_links_type &theSignalProvenance, 
+    std::vector<TotemRPDigi> &output_digi, 
+    simRP::DigiPrimaryMapType &output_digi_links) 
+{
+  for(simRP::strip_charge_map::const_iterator i=signals.begin(); 
+        i!=signals.end(); ++i)
+  {
+    //one threshold per hybrid
+    unsigned short strip_no = i->first;
+    if(i->second > threshold_ && (!dead_strips_simulation_on_ 
+          || dead_strips_.find(strip_no)==dead_strips_.end() ))
+    {
+      output_digi.push_back(TotemRPDigi(strip_no));
+      if(links_persistence_)
+      {
+        output_digi_links.push_back(theSignalProvenance[strip_no]);
+        if(verbosity_)
+        {
+          edm::LogInfo("RPVFatSimulator")<<" digi links size="<<theSignalProvenance[strip_no].size()<<"\n";
+          for(unsigned int u=0; u<theSignalProvenance[strip_no].size(); ++u)
+          {
+            edm::LogInfo("RPVFatSimulator")<<" digi: particle="<<theSignalProvenance[strip_no][u].first<<" energy [electrons]="<<theSignalProvenance[strip_no][u].second<<"\n";
+          }
+        }
+      }
+    }
+  }
+  
+  if(verbosity_)
+  {
+    for(unsigned int i=0; i<output_digi.size(); ++i)
+    {
+      edm::LogInfo("RPVFATSimulator")
+          <<output_digi[i].getStripNumber()<<"\n";
+    }
+  }
+}
+
+void RPVFATSimulator::SetDeadStrips()
+{
+  dead_strips_.clear();
+  double dead_strip_number = gRandom->Binomial(strips_no_, dead_strip_probability_);
```

Comment:
Is this test code or part of the actual simulation?

Reply:
The same as before (the sim. developer need to be contacted) @fabferro

Reply:
Sorry but I don't understand what you refer to with "before". Anyway, if this is used for production, the engine given by RandomNumberGenerator service should be used instead of `gRandom`.

Reply:
Sorry to be too short on this... We meant that as responsible for the integration of the sensor simulation (timing detector, tracking - strips and pixel), we don't know in advance the impact of removing this in the sensor simulation itself. We forwarded the comments to someone who might know it better (@fabferro @jan-kaspar @grzanka)

Reply:
So, we got input from the sensor simulator expert, removed the method (which was intended to simulate dead channels in the old time when TOTEM was not integrated into CMS), tested the simulation and it worked. So we decided to just remove it.